lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -222,47 +234,47 @@ def _from_file(self, root_path: Path, file_path: Path) -> List[dict]:
                 rule_file = yaml.safe_load(rf)
 
             except Exception as e:
-                logger.error("Failed to read alert rules from %s: %s", file_path.name, e)
+                logger.error("Failed to read rules from %s: %s", file_path.name, e)
                 return []
 
-            if _is_official_alert_rule_format(rule_file):
-                alert_groups = rule_file["groups"]
-            elif _is_single_alert_rule_format(rule_file):
-                # convert to list of alert groups
+            if _is_official_rule_format(rule_file):
+                groups = rule_file["groups"]
+            elif _is_single_rule_format(rule_file, self.rule_type):
```

Comment:
I may have missed a discussion on this: should we add a deprecation warning for the custom format?

This really feels like it should be an abstract method isn't it?

Reply:
Are we planning to deprecate it? AFAIK, no. At least nothing firm. @simskij likes it, anyway.

Since there's no _real_ difference in parsing alerts/recording rules (other than "no annotations" and what's in the YAML, neither of which we're enforcing right now), there's no reason to make this an abstract method where both concrete implementations would be identical.

The point is rather to avoid that, but if we feel like there's some reason to in the future, that we _can_, from a single, unified place.

This patch takes a middle ground between the discussion of "mix-n-match rules" (which will be hard or impossible to untangle in relation data, charm filesystems, and so on later) and separation, but saying "the actual _handling_ of the rules is more or less identical now, so let's make the code variadic enough to give the right messages back and not change filenames, but otherwise keep the same logic"). 

From that point of view, it's not clear at all where concrete implementations of abstract methods would differ, or why.